src/block.rs

```diff
@@ -49,10 +69,15 @@ impl Block {
     }
 
     /// Returns the `Block` hash
-    pub fn clone_hash(&self) -> [u8; 32] {
+    pub fn clone_hash(&self) -> Sha256Hash {
```

Comment:
I think it's better and consistent if hash() returns &self.hash and if cloning is necessary, user should do block.hash().clone().

Reply:
This merge is messed up. hold on.